docs/inventory_object.md

```diff
@@ -0,0 +1,194 @@
+# inventory directive in kustomization.yaml
+
+New in v2.1.0, a kustomization file may have an `inventory` field:
+```yaml
+inventory:
+  type: ConfigMap
+  configMap:
+    name: prune-cm-name
+    namespace: some-namespace
+```
+
+### Motivation
+If present, `Kustomize build` will make an _inventory_ object,
+which  could be a ConfigMap, or an App(to be added),
+ which can be consumed by a client such as those under development in
+[cli-experimental](https://github.com/kubernetes-sigs/cli-experimental).
+The client can recognize this object by name and use it to do a better job
+with actions like `apply`, `prune` and `delete`.
+
+Read more about this below.
+
+### Implementation
+
+The _inventory_ ConfigMap contains the following in its `data` field:
+
+- all objects within this kustomization target
+
+  Each record is a key-value pair in the data field.
+  The key represent the Group, Kind, Namespace and Name of an object.
+  The value is a hash computed based on current kustomization target.
+  ```yaml
+  <Group>_<Kind>_<Namespace>_<Name>: <hash>
+  ```
+  For example, a Deployment object can be represented as
+  ```yaml
+  apps_Deployment_default_mysql: m99cb8h85h
+  ```
+
+- all object reference within this kustomization target
+
+  Each record is also a key-value pair in the data field.
+  The key represents two objects, where one is referred by the other.
+  The value is a hash computed based on current kustomization target.
+  ```yaml
+  <Group1>_<Kind1>_<Namespace1>_<Name1>---<Group2>_<Kind2>_<Namespace2>_<Name2>: <hash>
+  ```
+  For example, a ConfigMap object referred by a Deployment object
+  can be represented as
+  ```yaml
+  _Secret_default_pass-ktm4b8f895---apps_Deployment_default_mysql:  m99cb8h85h
+  ```
+
+Basically, this ConfigMap object acts a record of objects that are applied as a group.
+This ConfigMap object can be consumed by a client such as
+[cli-experimental](https://github.com/kubernetes-sigs/cli-experimental).
+The client can recognize this ConfigMap object and take proper actions
+when running apply, prune and delete.
```

Comment:
or:

The inventory is a record of all the objects emitted by the `kustomize build` process.  If stored somewhere (i.e. in the cluster like any other k8s object), it can be compared to _newer_ inventory objects to see what changed and take action on those changes - like pruning Secrets that are no longer referenced.